src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AlleleSpecificAnnotationData.java

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import htsjdk.variant.variantcontext.Allele;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A class to encapsulate the raw data for allele-specific classes compatible with the ReducibleAnnotation interface
+ * @param <T> the type of raw data to be stored for later annotation calculation
+ */
+public final class AlleleSpecificAnnotationData<T> extends ReducibleAnnotationData<T>{
+    private final List<Allele> alleleList;
+    private final Allele refAllele;
+
+    public AlleleSpecificAnnotationData(final List<Allele> inputAlleles, final String inputData) {
+        super(inputData);
+        this.attributeMap = new HashMap<>();
+        inputAlleles.forEach(a -> {attributeMap.put(a, null);});
+        alleleList = inputAlleles;
+        refAllele = alleleList.stream().filter(Allele::isReference).findAny().orElse(null);
```

Comment:
This streaming expression is much less readable/straightforward than the previous for loop.